packages/proteus/src/main/keys/KeyPair.ts

```diff
@@ -79,26 +79,24 @@ export class KeyPair {
     this.secret_key.encode(encoder);
 
     encoder.u8(1);
-    return this.public_key.encode(encoder);
+    this.public_key.encode(encoder);
+
+    return encoder;
   }
 
   static decode(decoder: CBOR.Decoder): KeyPair {
-    const self = new KeyPair();
-
     const properties = decoder.object();
-    for (let index = 0; index <= properties - 1; index++) {
-      switch (decoder.u8()) {
-        case 0:
-          self.secret_key = SecretKey.decode(decoder);
-          break;
-        case 1:
-          self.public_key = PublicKey.decode(decoder);
-          break;
-        default:
-          decoder.skip();
-      }
+
+    if (properties === 2) {
+      decoder.u8();
+      const secretKey = SecretKey.decode(decoder);
+
+      decoder.u8();
+      const publicKey = PublicKey.decode(decoder);
+
+      return new KeyPair(publicKey, secretKey);
     }
 
-    return self;
+    throw new Error(`Unexpected number of properties: "${properties}"`);
```

Comment:
Let's use a `DecodeError` here, shall we?

Reply:
Great idea.